src/storage/mutate/UpdateVertexProcessor.cpp

```diff
@@ -128,21 +133,66 @@ kvstore::ResultCode UpdateVertexProcessor::collectVertexProps(
         }
     } else if (insertable_ && updateTagIds_.find(tagId) != updateTagIds_.end()) {
         resp_.set_upsert(true);
-        const auto constSchema = this->schemaMan_->getTagSchema(this->spaceId_, tagId);
-        if (constSchema == nullptr) {
+        const auto schema = this->schemaMan_->getTagSchema(this->spaceId_, tagId);
+        if (schema == nullptr) {
+            LOG(ERROR) << "Get nullptr schema";
+            return kvstore::ResultCode::ERR_UNKNOWN;
+        }
+        auto tagStatus = this->schemaMan_->toTagName(this->spaceId_, tagId);
+        if (!tagStatus.ok()) {
+            LOG(ERROR) << "toTagName failed, tagId: " << tagId;
             return kvstore::ResultCode::ERR_UNKNOWN;
         }
-        for (auto index = 0UL; index < constSchema->getNumFields(); index++) {
-            auto propName = std::string(constSchema->getFieldName(index));
-            OptVariantType value = RowReader::getDefaultProp(constSchema.get(), propName);
+        auto tagName = std::move(tagStatus.value());
+
+        auto updater = std::make_unique<RowUpdater>(schema);
+        // When the schema field is not in update field or the update item has src item,
+        // need to get default value from schema. If nonexistent return error.
+        for (auto index = 0UL; index < schema->getNumFields(); index++) {
+            auto propName = std::string(schema->getFieldName(index));
+            auto findIter = std::find_if(updateItems_.cbegin(), updateItems_.cend(),
+                    [&propName, &tagName](auto &item) {
+                return item.prop == propName && item.name == tagName; });
+            OptVariantType value;
+
+            if (findIter == updateItems_.end()) {
+                // When the schema field is not in update field
+                // need to get default value from schema. If nonexistent return error.
+                value = schema->getDefaultValue(index);
+            } else {
+                // When the update item has src item,
+                // need to get default value from schema. If nonexistent return error.
```

Comment:
why to get updated field default value.
I think it don't need

Reply:
Because insert when upsert,  filter condition is alway true

Reply:
Yeah, when the PR submitted, it didn't say when upsert was not valid

Reply:
insert when upsert， filter is always true.

Here you get default value of updated field,  Need to be updated later.

so here getting default value of updated field is redundancy